external/avro/src/main/scala/org/apache/spark/sql/avro/functions.scala

```diff
@@ -45,9 +45,10 @@ object functions {
   }
 
   /**
-   * Converts a binary column of avro format into its corresponding catalyst value. The specified
-   * schema must match the read data, otherwise the behavior is undefined: it may fail or return
-   * arbitrary result.
+   * Converts a binary column of avro format into its corresponding catalyst value. If a writer's
+   * schema is provided in the options, a different (but compatible) schema can be used for reading.
```

Comment:
If the actual Avro schema is provided via the option `actualSchema`

Reply:
Good point, thanks